DependencyInjection/GraphQLiteCompilerPass.php

```diff
@@ -479,34 +479,18 @@ private function getCodeCache(): ClassBoundCacheContractInterface
      * Returns the array of globbed classes.
      * Only instantiable classes are returned.
      *
-     * @return array<string,ReflectionClass<object>> Key: fully qualified class name
+     * @param string $namespace
+     * @return array<class-string, ReflectionClass<object>>|Generator Key: fully qualified class name
      */
-    private function getClassList(string $namespace, int $globTtl = 2, bool $recursive = true): array
+    private function getClassList(string $namespace): array|Generator
     {
-        $explorer = new GlobClassExplorer($namespace, $this->getPsr16Cache(), $globTtl, ClassNameMapper::createFromComposerFile(null, null, true), $recursive);
-        $allClasses = $explorer->getClassMap();
-        $classes = [];
-        foreach ($allClasses as $className => $phpFile) {
-            if (! class_exists($className, false)) {
-                // Let's try to load the file if it was not imported yet.
-                // We are importing the file manually to avoid triggering the autoloader.
-                // The autoloader might trigger errors if the file does not respect PSR-4 or if the
-                // Symfony DebugAutoLoader is installed. (see https://github.com/thecodingmachine/graphqlite/issues/216)
-                require_once $phpFile;
-                // @phpstan-ignore-next-line Does it exist now?
-                if (! class_exists($className, false)) {
-                    continue;
-                }
-            }
-
-            $refClass = new ReflectionClass($className);
-            if (! $refClass->isInstantiable()) {
+        $finder = new ComposerFinder();
+        foreach ($finder->inNamespace($namespace) as $class) {
+            if (!$class instanceof ReflectionClass){
```

Comment:
We can use `subclassOf` from kcs/class-finder.

But I'm not sure this is the correct logic, in fact previously we were just returning all the classes in the namespace along with their reflection. 

It is unclear to me that the `$reflector` from kcs/class-finder actually is a ReflectionClass.

Reply:
According to docs finder returns Reflector which in 100% of times is ReflectionClass but to be sure (and make phpstan happy) this check handles it, I made the same thing in graphqlite and covered it with test and it passes. Anyway if you sure with subclass I can change it.

Reply:
Okay so we do not want a "subclass" but an `assert` should be sufficient if this is guaranteed to be a `ReflectionClass`.

Alternatively it should be possible to update the return type in the library.